packages/create-block/lib/scaffold.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+const { dirname, join } = require( 'path' );
+const makeDir = require( 'make-dir' );
+const { readFile, writeFile } = require( 'fs' ).promises;
+const { render } = require( 'mustache' );
+
+/**
+ * Internal dependencies
+ */
+const initWPScripts = require( './init-wp-scripts' );
+const { code, clear, info, success } = require( './log' );
+const { hasWPScriptsEnabled, getOutputFiles } = require( './templates' );
+
+module.exports = async function( templateName, {
+	namespace,
+	slug,
+	title,
+	description,
+	dashicon,
+	category,
+	author,
+	license,
+	version,
+} ) {
+	clear();
+	info( `Creating a new WordPress block in "${ slug }" folder.` );
+
+	const outputFiles = getOutputFiles( templateName );
+	const view = {
+		namespace,
+		slug,
+		machineName: `${ namespace }_${ slug }`.replace( /\-/g, '_' ),
```

Comment:
`machineName` was a bit unclear to me in its meaning until I saw how it was used for generating PHP functions.

My first instinct was that we don't need it if we can just give the templates the namespace and slug to use, but that's not possible because they need the snake-case.

My second instinct was that we could provide `namespaceSnakeCase` and `slugSnakeCase` as separate properties.

My third instinct is that we could provide some utility functions for the templates. I'm familiar with these moreso in Handlebars (the `{{name|snakeCase}}` syntax), but it might be something that Mustache makes possible through its "Functions"?

https://www.npmjs.com/package/mustache#functions

Something like: `{{#snakeCase}}{{namespace}}{{/snakeCase}}_{{#snakeCase}}{{slug}}{{/snakeCase}}`

(Which I recognize now is not all that readable)

I don't really feel strongly about any of these alternatives, so don't feel compelled to make a change here. Just sharing some of my thought process in reacting to this.

Reply:
`machineName` was ported from WP CLI scaffold where it is an established pattern in templates. I'm fine with changing this name to better match with JavaScript 👍 

`namespaceSnakeCase` and `slugSnakeCase` sounds good. I will go with this recommendation.

Reply:
One observation. I was surprised seeing how lodash handles `es5-example`. It converts it to:
- `snakeCase` -> `es_5_example`
- `startCase` -> `ES 5 Example`